### PR TITLE
Compress the report output when running make functests

### PR DESCRIPTION
**Pull request by fedepaol**
This should save some space on CI

### CODE REVIEW

File: hack/run-functests.sh [hunk 1]
```diff
@@ -39,6 +39,15 @@ else
   EXEC_TESTS="cnf-tests/test-run.sh -ginkgo.focus=$FOCUS -junit $TESTS_REPORTS_PATH -report $TESTS_REPORTS_PATH"
 fi
 
+reports="cnftests_failure_report.log setup_failure_report.log validation_failure_report.log"
+
+for report in $reports; do
+  if [[ -f "$TESTS_REPORTS_PATH/$report" ]]; then    
+    gzip "$TESTS_REPORTS_PATH/$report"
```
Comment (SchSeba) [inline hack/run-functests.sh:46]: do we have gzip inside the cnf container?

File: hack/run-functests.sh [hunk 2]
```diff
@@ -39,6 +39,15 @@ else
...
+    mv "$TESTS_REPORTS_PATH/$report".gz "$TESTS_REPORTS_PATH/$report.""$(date +"%Y-%m-%d_%T")".gz
```
Comment (SchSeba) [inline hack/run-functests.sh:47]: do we also want to remove the files?

File: hack/run-functests.sh (hunk 1 above)
Comment (fedepaol) [reply hack/run-functests.sh:46]: That is happening outside the container (the user's terminal OR the pod that is running CI tests).
On CI side, will find out :P

File: hack/run-functests.sh (hunk 2 above)
Comment (fedepaol) [reply hack/run-functests.sh:47]: gzip already removes the original file

File: hack/run-functests.sh (hunk 1 above)
Comment (fedepaol) [reply hack/run-functests.sh:46]: Just ssh-ed into the test runner pod, it has it onboard

**Pull request merged**